### PR TITLE
Avoid alignment in FilePrefetchBuffer during seek with async_io

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: During Seek, the iterator seeks every file on L0. In async_io, it submit the requests to seek on every file on L0 asynchronously using RocksDB FilePrefetchBuffer.
However, FilePrefetchBuffer does alignment and reads extra bytes then needed that can increase the throughput. 
In case of non direct io, the alignment can be avoided.

Test Plan: 
- Added a unit test that fails without this PR.
- make crash_test -j32 completed successfully

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by akankshamahajan15** [comment]
`build-linux-unity-and-headers` job  failing on main branch

**Comment by akankshamahajan15** [comment]
`crashtest` failing on the main branch.

**Review by anand1976 — COMMENTED** [review]
LGTM, with one minor comment. Can we add a unit test for this case?

**Comment by akankshamahajan15** [comment]
> LGTM, with one minor comment. Can we add a unit test for this case?

Yes. I'm working on adding unit tests. I'll update the summary once they complete.

File: unreleased_history/bug_fixes/opt_seek.md [hunk 1]
```diff
@@ -0,0 +1 @@
+During seek, the iterator seeks every file on L0. In async_io, it submit the requests to seek on every file on L0 asynchronously using RocksDB FilePrefetchBuffer. However, FilePrefetchBuffer does alignment and reads extra bytes then needed that can increase the throughput. Added a fix to avoid that alignment when direct_io is disabled.
```
Comment (anand1976) [inline unreleased_history/bug_fixes/opt_seek.md:1]: "Every file on L0, and every non-zero level"

Comment (anand1976) [inline unreleased_history/bug_fixes/opt_seek.md:1]: Type - "extra bytes then needed" -> "extra bytes than needed"

Comment (anand1976) [reply unreleased_history/bug_fixes/opt_seek.md:1]: I think we can omit the detail of seeking every file in L0 and other levels. Just say that when reading a block for seeking a target key in a file and not doing readahead, the iterator was aligning the read on a page boundary and reading more than necessary. This increases the storage read bandwidth usage.

**Review by anand1976 — APPROVED** [review]
A couple of suggestions for describing the fix in unreleased_history.

**Pull request closed without merging**